pkg/operator/ceph/object/rgw.go

```diff
@@ -269,14 +275,20 @@ func (c *clusterConfig) deleteLegacyDaemons() {
 func (c *clusterConfig) deleteStore() error {
 	logger.Infof("deleting object store %q from namespace %q", c.store.Name, c.store.Namespace)
 
-	// Delete rgw CephX keys
+	// Delete rgw CephX keys and configuration in centralized mon database
 	for i := 0; i < int(c.store.Spec.Gateway.Instances); i++ {
 		daemonLetterID := k8sutil.IndexToName(i)
 		depNameToRemove := fmt.Sprintf("%s-%s-%s", AppName, c.store.Name, daemonLetterID)
 		err := cephclient.AuthDelete(c.context, c.store.Namespace, generateCephXUser(depNameToRemove))
 		if err != nil {
 			return err
 		}
+
+		// Delete configuration in centralized mon database
+		err = c.deleteFlagsMonConfigStore(depNameToRemove)
```

Comment:
I wonder if this shouldn't happen before we remove the auth key.

Reply:
I will change the order but I'm not sure it really matters